### PR TITLE
Add PCI expansion ROM support

### PR DESCRIPTION
**Pull request by ElectrodeYT**

### CODE REVIEW

File: kernel/thor/system/pci/thor-internal/pci/pci.hpp [hunk 1]
```diff
@@ -313,6 +320,7 @@ struct PciDevice final : PciEntity {
 
 	// device configuration
 	PciBar bars[6];
+	PciExpansionRom expansion_rom;
```
Comment (no92) [inline kernel/thor/system/pci/thor-internal/pci/pci.hpp:323]: class members should be camelCased.

**Review by no92 — APPROVED** [review]
LGTM other than 1 style nitpick.

File: kernel/thor/system/pci/pci_discover.cpp [hunk 2]
```diff
@@ -167,6 +172,29 @@ namespace {
 			assert(headError == Error::success);
 			assert(tailError == Error::success);
 
+			auto descError = co_await PushDescriptorSender{conversation, std::move(descriptor)};
+			// TODO: improve error handling here.
+			assert(descError == Error::success);
+		}else if(preamble.id() == bragi::message_id<managarm::hw::AccessExpansionRomRequest>) {
```
Comment (Dennisbonke) [inline kernel/thor/system/pci/pci_discover.cpp:178]: Missing space between `}` and the `else`.

File: kernel/thor/system/pci/pci_discover.cpp [hunk 3]
```diff
@@ -167,6 +172,29 @@ namespace {
...
+			auto req = bragi::parse_head_only<managarm::hw::AccessExpansionRomRequest>(reqBuffer, *kernelAlloc);
+
+			if (!req) {
```
Comment (Dennisbonke) [inline kernel/thor/system/pci/pci_discover.cpp:181]: We don't do spaces between `if`and the opening brace.

**Review by Dennisbonke — APPROVED** [review]
Two style issues, otherwise LGTM

**Pull request merged**